modules/porous_flow/src/dirackernels/PorousFlowSquarePulsePointSource.C

```diff
@@ -0,0 +1,80 @@
+/****************************************************************/
+/* MOOSE - Multiphysics Object Oriented Simulation Environment  */
+/*                                                              */
+/*          All contents are licensed under LGPL V2.1           */
+/*             See LICENSE for full restrictions                */
+/****************************************************************/
+
+#include "PorousFlowSquarePulsePointSource.h"
+
+template<>
+InputParameters validParams<PorousFlowSquarePulsePointSource>()
+{
+  InputParameters params = validParams<DiracKernel>();
+  params.addRequiredParam<Real>("mass_flux", "The mass flux at this point in kg/s (positive is flux in, negative is flux out)");
+  params.addRequiredParam<std::vector<Real> >("point", "The x,y,z coordinates of the point");
+  params.addParam<Real>("start_time", 0.0, "The time at which the source will start");
+  params.addParam<Real>("end_time", 1.0e30, "The time at which the source will end");
+  return params;
+}
+
+PorousFlowSquarePulsePointSource::PorousFlowSquarePulsePointSource(const InputParameters & parameters) :
+    DiracKernel(parameters),
+    _mass_flux(getParam<Real>("mass_flux")),
+    _point_param(getParam<std::vector<Real> >("point")),
+    _start_time(getParam<Real>("start_time")),
+    _end_time(getParam<Real>("end_time"))
+{
+  _p(0) = _point_param[0];
+
+  if (_point_param.size() > 1)
+    _p(1) = _point_param[1];
+  if (_point_param.size() > 2)
+      _p(2) = _point_param[2];
+}
+
+void
+PorousFlowSquarePulsePointSource::addPoints()
+{
+  addPoint(_p);
+}
+
+Real
+PorousFlowSquarePulsePointSource::computeQpResidual()
+{
+  Real factor=.0, rate = 0.;
```

Comment:
operator spacing. My personal preference is to specify digits before _and_ after the decimal point for Reals (`0.0`). But at least make it consistent.